### PR TITLE
Fix OIDC flags

### PR DESCRIPTION
**Pull request by richabanker**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Fixes the initialization of OIDC related flags

#### Which issue(s) this PR fixes:

Fixes issue regarding OIDC flags initialization described [here](https://github.com/kubernetes/kubernetes/pull/127581#issuecomment-2463010854)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sttts** [comment]
Lgtm.

Needs somebody from sig-auth to review, @liggitt @enj

File: pkg/kubeapiserver/options/authentication.go [hunk 1]
```diff
@@ -888,3 +870,12 @@ func loadAuthenticationConfigFromData(data []byte) (*apiserver.AuthenticationCon
 
 	return configuration, nil
 }
+
+// areFlagsConfigured is a function that returns true if any of the oidc-* flags are configured.
+func (o *OIDCAuthenticationOptions) areFlagsConfigured() bool {
+	return !reflect.DeepEqual(o, NewBuiltInAuthenticationOptions().WithOIDC())
```
Comment (liggitt) [inline pkg/kubeapiserver/options/authentication.go:876]: this isn't equivalent, this wouldn't catch someone explicitly setting a flag to the default

Comment (liggitt) [reply pkg/kubeapiserver/options/authentication.go:876]: fs.Changed returns true if Set was called, even if it was set to the same value

Comment (richabanker) [reply pkg/kubeapiserver/options/authentication.go]: Hm, right. Reverted the changes done around `areFlagsSet`

Comment (sttts) [reply pkg/kubeapiserver/options/authentication.go]: There are also enough repackaging of kube-apiserver like k3s, microshift etc. It's not guaranteed that the values actually come from real flags.

Comment (liggitt) [reply pkg/kubeapiserver/options/authentication.go]: > Hm, right. Reverted the changes done around `areFlagsSet`

the change now looks ~fine, does it actually resolve the oidc integration test initialization issue you were seeing?

Comment (richabanker) [reply pkg/kubeapiserver/options/authentication.go]: Yes the test is passing with these changes + the flagz changes.

**Comment by liggitt** [comment]
/lgtm
/approve

This change is strictly _more_ correct (making AddFlags non-mutating for the public fields). I agree the `areFlagsConfigured` field is wonky and would be glad to see it untangled in a follow-up if possible.

**Pull request merged**